### PR TITLE
Finally remove wxPython!!

### PR DESCRIPTION
**Pull request by bzier**
I finally got around to playing around with removing the `wxPython` dependency, and was able to successfully remove it!!!

## GUI
I discovered [Tkinter](https://wiki.python.org/moin/TkInter) which is Python's 'default' GUI package. It is built-in and supposedly cross-platform. I re-wrote the `record.py` GUI to use that library instead of `wx`. 

## Screenshots
`wx` was also being used for the screenshots, which I replaced with `mss`, as I had done for the `gym-mupen64plus` project.

## Performance
Additionally, I made a slight performance enhancement (unrelated to `wx`). The `.csv` file was being opened, written, flushed, and closed on every cycle. I did a little measurement and found that moving the overhead of the open & flush/close to happen once (at the beginning and ending of a recording session) improved performance.

Here are the numbers for cumulative open (O), write (W), and close (C) times for 50 cycles. I ran the test 3 times for each the old/new code (notice the first test is slower for both versions).

### Old code (open/close every cycle):
```
O: 0.0103859901428
W: 0.0031590461731
C: 0.00543642044067
Total time 0.0189814567566

O: 0.0020763874054
W: 0.00172877311707
C: 0.000882863998413
Total time 0.00468802452087

O: 0.0020866394043
W: 0.00154590606689
C: 0.000622987747192
Total time 0.00425553321838
```

### New code (open/close once per recording):
```
O: 5.50746917725e-05
W: 0.00775456428528
C: 3.19480895996e-05
Total time 0.00784158706665

O: 9.20295715332e-05
W: 0.00163722038269
C: 4.79221343994e-05
Total time 0.00177717208862

O: 6.38961791992e-05
W: 0.0017421245575
C: 4.19616699219e-05
Total time 0.00184798240662
```

As you can see, the total time required for the new code's file access to the `.csv` file was anywhere from half to a quarter of the time required with the old code. And this is limited to 50 cycles (screenshots & controller samples). This difference would increase with longer recordings.

## Related
Towards #53 but only tested on Linux (Ubuntu 16.04). May still have issues with cross-platform, but needs to be confirmed/tested.

### CODE REVIEW

File: record.py [hunk 1]
```diff
@@ -31,54 +40,47 @@ def __init__(self):
...
+        bottom_half = tk.Frame(self.root)
+        bottom_half.pack(side=tk.LEFT, padx=5, pady=10)
+        
         # Images
-        img = wx.Image(320,240)
-        self.image_widget = wx.StaticBitmap(self.img_panel, wx.ID_ANY, wx.Bitmap(img))
-
+        self.img_panel = tk.Label(top_half, image=ImageTk.PhotoImage("RGB", size=IMAGE_SIZE)) # Placeholder
+        self.img_panel.pack(side = tk.LEFT, expand=False, padx=5)
+
```
Comment (kevinhughes27) [inline record.py]: extra whitespace?

File: record.py [hunk 2]
```diff
@@ -31,54 +40,47 @@ def __init__(self):
...
-        self.img_panel = wx.Panel(self)
-        self.joy_panel = wx.Panel(self)
-        self.record_panel = wx.Panel(self)
-
+        top_half = tk.Frame(self.root)
+        top_half.pack(side=tk.TOP, expand=True, padx=5, pady=5)
+        message = tk.Label(self.root, text="(Note: UI updates are disabled while recording)")
+        message.pack(side=tk.TOP, padx=5)
+        bottom_half = tk.Frame(self.root)
+        bottom_half.pack(side=tk.LEFT, padx=5, pady=10)
```
Comment (kevinhughes27) [inline record.py]: if I'm understanding right this is aligned vertically now? Can you include a screenshot?

File: record.py [hunk 3]
```diff
@@ -87,28 +89,38 @@ def init_plot(self):
...
-        return bmp
+        with mss.mss() as sct:
+            # Get raw pixels from the screen
+            sct_img = sct.grab({   "top": Screenshot.OFFSET_X,
+                                  "left": Screenshot.OFFSET_Y,
+                                 "width": Screenshot.SRC_W,
+                                "height": Screenshot.SRC_H})
+
+            # Create the Image
+            return Image.frombytes('RGB', sct_img.size, bytes(sct_img.raw), 'raw', 'BGRX')
```
Comment (BoboTiG) [inline record.py]: :+1:

File: record.py [hunk 4]
```diff
@@ -87,28 +89,38 @@ def init_plot(self):
...
+            self.t += 1
+
 
     def take_screenshot(self):
-        screen = wx.ScreenDC()
-        bmp = wx.Bitmap(Screenshot.SRC_W, Screenshot.SRC_H)
-        mem = wx.MemoryDC(bmp)
-        mem.Blit(0, 0, Screenshot.SRC_W, Screenshot.SRC_H, screen, Screenshot.OFFSET_X, Screenshot.OFFSET_Y)
-        return bmp
+        with mss.mss() as sct:
```
Comment (BoboTiG) [inline record.py]: For better performances, you could assign it to a class attribute. With MSS version 3.2.0 (just released), I benchmarked and it apperas that avoiding instanciation is a **huge** improvement.

File: record.py (hunk 2 above)
Comment (bzier) [reply record.py]: Kinda, I have a vertical stack of three things:
- the frame containing the image and joystick graph (side-by-side)
- a message label informing if the lack of UI updates during recording
- the frame containing the directory textbox and record button (side-by-side)

Essentially it should look just like it did before (with the added message). I'll post a screenshot in a bit.

File: record.py (hunk 1 above)
Comment (bzier) [reply record.py]: Oops, good catch. I think I have VS Code ignoring whitespace in diffs. I'll clean it up soon.

File: record.py (hunk 4 above)
Comment (bzier) [reply record.py]: Ah yes, thanks for pointing that out. I think I read that yesterday, but forgot to move it. In my other project I'm just instantiating once and holding onto the reference. I'll do the same here in a bit.

File: record.py (hunk 2 above)
Comment (bzier) [reply record.py]: record_setup

Comment (bzier) [reply record.py]: I also updated the image in the screenshots directory with this one.

File: record.py (hunk 4 above)
Comment (bzier) [reply record.py]: Wow, you weren't kidding about it ***just*** being released. Quite literally just before you posted your comment. And you've already added this project to the list. You really don't waste any time :)

Comment (bzier) [reply record.py]: I've now upgraded to `3.2.0`, utilized the `.bgra` property, and moved instantiation to the `__init__` method so we only do it once. Thanks for the comment :+1:

Comment (BoboTiG) [reply record.py]: Nice! :)

Comment (kevinhughes27) [reply record.py]: This thread makes me so happy! Thanks for keeping us in the loop about updates and changes to mss :heart::heart::heart:

File: record.py (hunk 2 above)
Comment (kevinhughes27) [reply record.py]: Sweet! I saw the `tk.Left` after commenting and figured you probably made it more or less the same. Thanks for the screenshot it looks great.

**Pull request merged**